openml/runs/run.py

```diff
@@ -186,24 +192,45 @@ def _generate_arff_dict(self):
         class_labels = task.class_labels
 
         arff_dict = OrderedDict()
-        arff_dict['attributes'] = [('repeat', 'NUMERIC'),  # lowercase 'numeric' gives an error
-                                   ('fold', 'NUMERIC'),
-                                   ('sample', 'NUMERIC'),
-                                   ('row_id', 'NUMERIC')] + \
-            [('confidence.' + class_labels[i], 'NUMERIC') for i in range(len(class_labels))] +\
-            [('prediction', class_labels),
-             ('correct', class_labels)]
         arff_dict['data'] = self.data_content
         arff_dict['description'] = "\n".join(run_environment)
-        arff_dict['relation'] = 'openml_task_' + str(task.task_id) + '_predictions'
+        arff_dict['relation'] = 'openml_task_' + str(task.task_id) + \
+                                '_predictions'
+
+        # Separate these out? Normal classification doesn't need 'sample'
+        if task.task_type in ['Supervised Classification', 'Learning Curve']:
```

Comment:
please see comment. In case we separate this, also make a special one for learning curve and classification (classification does not need sample)

Reply:
Additionally, I am not convinced about this whole if else statement here. I can of feel like we want to factorize this to some canonical place. Will come back to this when I do my final review :)

Reply:
I didn't create this if-else structure. It was there before. I just added one case because I didn't want to upset the code structure too much here. I'm not sure what you want to say with 'some canonical place'.

Reply:
@janvanrijn do you maybe want to say that these should be in a task-specific file so that in order to add a new task the user does not have to touch `run.py`? Could this information be added to the task class?